packages/apps/graph/src/graph/objects/Transfer.ts

```diff
@@ -17,6 +17,68 @@ export default builder.prismaNode('Transfer', {
     requestKey: t.exposeString('requestKey'),
     receiverAccount: t.exposeString('receiverAccount'),
 
+    // computed fields
+    crossChainTransfer: t.prismaField({
+      type: 'Transfer',
+      nullable: true,
+      async resolve(__query, parent) {
+        try {
+          let counterTransaction;
+
+          // Try to find finisher transfer
+          const finisherTransfer = await prismaClient.transaction.findFirst({
+            where: {
+              pactId: parent.requestKey,
+            },
+          });
+
+          if (finisherTransfer) {
+            counterTransaction = finisherTransfer;
+          } else {
+            // If not found, try to find the initiating transfer
+            // First find the corresponding transaction
+            const finisherTransaction =
+              await prismaClient.transaction.findFirst({
+                where: {
+                  blockHash: parent.blockHash,
+                  requestKey: parent.requestKey,
+                },
+              });
+
+            if (
+              !finisherTransaction ||
+              finisherTransaction.pactId === undefined ||
+              finisherTransaction.pactId === null
+            ) {
+              return null;
+            }
+
+            // Then find the initiating transaction with the pactId
+            const initiatingTransaction =
+              await prismaClient.transaction.findFirstOrThrow({
+                where: {
+                  requestKey: finisherTransaction.pactId,
+                  pactId: undefined,
+                },
+              });
+
+            counterTransaction = initiatingTransaction;
+          }
+
+          return await prismaClient.transfer.findFirst({
+            where: {
+              blockHash: counterTransaction.blockHash,
+              requestKey: counterTransaction.requestKey,
+              OR: [{ senderAccount: '' }, { receiverAccount: '' }],
+              amount: parent.amount,
+            },
+          });
+        } catch (error) {
+          throw normalizeError(error);
```

Comment:
Did you play around with the input params to see if you can generate errors that are not yet covered in the 'normalizeError' function? Or vague errors that could use a 'description' field in the error object.

Reply:
Yep the only error that could potentially occur is not finding the corresponding initiating transfer and that one logs correctly on the client. And for me to replicate this error I had to manually change data in the database. I dont think there are a lot of possible errors 

![image](https://github.com/kadena-community/kadena.js/assets/112877112/b0144431-1714-4523-9e07-51d64aadb53d)